compiler/rustc_builtin_macros/src/deriving/generic/ty.rs

```diff
@@ -69,6 +70,10 @@ impl Path {
             }
         }
     }
+
+    pub fn components(&self) -> &[Symbol] {
```

Comment:
```suggestion
    pub fn segments(&self) -> &[Symbol] {
```

I think we generally use the terminology *path segments* in the context of Rust paths contrary to filesystem paths which indeed use *path components*.